include/nbl/builtin/hlsl/concepts/core.hlsl

```diff
@@ -0,0 +1,110 @@
+// Copyright (C) 2024-2025 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+#ifndef _NBL_BUILTIN_HLSL_CONCEPTS_CORE_HLSL_INCLUDED_
+#define _NBL_BUILTIN_HLSL_CONCEPTS_CORE_HLSL_INCLUDED_
+
+
+#include <nbl/builtin/hlsl/concepts.hlsl>
+#include <nbl/builtin/hlsl/vector_utils/vector_traits.hlsl>
+#include <nbl/builtin/hlsl/type_traits.hlsl>
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace concepts
+{
+
+#ifdef __cpp_concepts // CPP
+
+#include <concepts>
+
+// Alias some of the std concepts in nbl. As this is C++20 only, we don't need to use
+// the macros here.
+template <typename T, typename U>
+concept same_as = std::same_as<T, U>;
+
+template <typename D, typename B>
+concept derived_from = std::derived_from<D, B>;
+
+template <typename F, typename T>
+concept convertible_to = std::convertible_to<F, T>;
+
+template <typename T, typename F>
+concept assignable_from = std::assignable_from<T, F>;
+
+template <typename T, typename U>
+concept common_with = std::common_with<T, U>;
+
+template <typename T>
+concept integral = std::integral<T>;
+
+template <typename T>
+concept signed_integral = std::signed_integral<T>;
+
+template <typename T>
+concept unsigned_integral = std::unsigned_integral<T>;
+
+template <typename T>
+concept floating_point = std::floating_point<T>;
+
+// Some other useful concepts.
+
+template<typename T, typename... Ts>
+concept any_of = (same_as<T, Ts> || ...);
+
+template <typename T>
+concept scalar = floating_point<T> || integral<T>;
```

Comment:
I'd want `is_scalar_v && ` added because of https://github.com/Devsh-Graphics-Programming/Nabla/pull/803/files#r1904299686

Reply:
done

Reply:
bump, the new code still only has `FloatingPoint<T> || Integral<T>`

Reply:
now its just `concept scalar = nbl::hlsl::is_scalar_v<T>;`